backend_inotify.go

```diff
@@ -121,6 +129,30 @@ type Watcher struct {
 	paths       map[int]string    // Map of watched paths (watch descriptor → path)
 	done        chan struct{}     // Channel for sending a "quit message" to the reader goroutine
 	doneResp    chan struct{}     // Channel to respond to Close
+
+	// Fanotify fields
+	// flags passed to fanotify_init
+	flags uint
+	// mount fd is the file descriptor of the mountpoint
+	mountpoint         *os.File
+	kernelMajorVersion int
+	kernelMinorVersion int
+	entireMount        bool
+	notificationOnly   bool
+	stopper            struct {
+		r *os.File
+		w *os.File
+	}
+	fanotify bool
+	// FanotifyEvents holds either notification events for the watched file/directory.
+	FanotifyEvents chan FanotifyEvent
+	// PermissionEvents holds permission request events for the watched file/directory.
+	//   fsnotify.PermissionToOpen    Permission request to open a file or directory. (Applicable only to fanotify watcher.)
+	//
+	//   fsnotify.PermissionToExecute Permission to open file for execution. (Applicable only to fanotify watcher.)
+	//
+	//   fsnotify.PermissionToRead    Permission to read a file or directory. (Applicable only to fanotify watcher.)
+	PermissionEvents chan FanotifyEvent
```

Comment:
I don't think we should have any references to fanotify in the backend_inotify.go file.

What I'd do for now is adjust the build tags on `backend_inotify.go` so it never compiles and then implement a new `Watcher` in `backend_fanotify.go`. This way all the tests should run against the fanotify backend.

I need to think a bit on how to best architecture things on this so you can tell you prefer to use fanotify; my idea was that you can select the backend per-watch.

Reply:
Sounds good. A step closer. Will update back soon.